### PR TITLE
Refactoring (part 1)

### PR DESCRIPTION
**Pull request by mjaehn**
### [Updating and consolidating Processing Chain for ICON-ART](https://github.com/C2SM/Tasks-Support/issues/12) (C2SM Task WG2 in 2023/1)

With the increasing number of supported model configurations and simulation options, parts of the processing chain (especially the main routine `run_chain.py`) have become rather complex and somewhat convoluted. 

The following parts of refactoring the code are addressed in this PR:

- [x] change naming: `target` -> `model`, `subtarget` -> `variant`
- [x] provide input files as dictionary (later to be replaced by `config.yaml` file)
- [x] introduce `models.yaml` file for global model configurations
- [x] reduce size of input data for test cases
- [x] job scripts: target models directly by their name
- [x] `run_chain.py`: distinguish features and variants instead of models
- [x] Remove non-test cases
- [x] add example test case for icon-art
- [x] add example test case for icon-art-global
- [x] update README.md

**This is part 1 of the refactoring task.** Part 2 will mainly consist of the following:

- Change case configuration files to a more "config-like" format: `config.py` -> `config.yaml`
- Docs: Revision, updating, publish to github.io
- Add icon-art-oem-ensemble test case
- Check if `variants` and `features` can be unified
- Further quality-of-life changes for more intuitive handling of the chain

### CODE REVIEW

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
launch jenkins

**Comment by mjaehn** [comment]
All tests are green. Ready to review 💡

File: config/models.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+models:
+    cosmo:
+        jobs: ['prepare_data', 'int2lm', 'cosmo', 'post_cosmo']
+        variants: ['spinup']
+        features: ['restart']
```
Comment (efmkoene) [inline config/models.yaml:5]: What is the difference between a job, a variant and a feature?

File: jobs/emissions.py [hunk 2]
```diff
@@ -80,8 +80,3 @@ def main(starttime, hstart, hstop, cfg):
             source_path = os.path.join(emis_dir, emis_prefix + filename_ending)
             dest_path = os.path.join(target_dir, dest_prefix + filename_ending)
             tools.copy_file(source_path, dest_path)
-
-            # convert grid_mapping_name from string (NF90_STRING) to char
-            # (NF90_CHAR) (needed for int2lm to work)
-            if cfg.target is tools.Target.COSMO:
-                tools.string2char.main(dest_path)
```
Comment (efmkoene) [inline jobs/emissions.py:87]: Did we lose some functionality, or was this not needed after all?

File: jobs/tools/write_cosmo_input_ghg.py [hunk 3]
```diff
@@ -107,8 +107,9 @@ def main(csv_filename, namelist_filename, cfg=None):
     """
 
     #Distinguish between COSMO and ICON
-    model = "COSMO"
-    if cfg.target is tools.Target.ICON or cfg.target is tools.Target.ICONART:
+    if cfg.model.startswith('cosmo'):
+        model = "COSMO"
+    if cfg.model.startswith('icon'):
         model = "ICON"
```
Comment (efmkoene) [inline jobs/tools/write_cosmo_input_ghg.py:113]: Does it happen that `write_cosmo_input_ghg` is called for ICON? Is it the correct function name in that case?

**Review by efmkoene — COMMENTED** [review]
Looks alright to me. Did I see correctly that there is no longer an ensembles testcase?

File: README.md [hunk 4]
```diff
@@ -16,28 +16,29 @@ instruction for the installation.
...
-$SCRATCH` and `pwd` at the command line to get to your personal scratch
-directory on Daint). When the command prompt asks for installation
-location, provide the path to your scratch and append `/miniconda3`.
+Install as user specific Miniconda, e.g. on your `$HOME` directory,
+which is the default location.
 
-> **Note**: The default location would be on your `$HOME` directory, which
-> may lead to memory issues.
+> **Note**: Only conda itself should be installed in your `$HOME`.
+> All environments should be stored in your `$PROJECT` directory,
```
[the remark this answers is not in the document]
Comment (mjaehn) [reply README.md:23]: This is general CSCS terminology: https://user.cscs.ch/storage/file_systems/
How come you are not assigned to a project?

File: cases/icon-art-global-test/config.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+# Initial conditios
+era5_inicond = False  # TO MODIFY
+species_inicond = True
+species2restart = ['TROH']
+
+# Nudging (meteorological and tracers)
+era5_global_nudging = False
+species_global_nudging = False
+species2nudge = []
```
[the remark this answers is not in the document]
Comment (mjaehn) [reply cases/icon-art-global-test/config.py:66]: @jthanwer What do you think?

File: cases/icon-art-oem-test/config.py [hunk 6]
```diff
@@ -5,55 +5,49 @@
...
-input_root_chem = '/store/empa/em05/input_iconart_processing_chain_example/chem'
-
+input_root_chem = os.path.join(input_root, 'chem')
 input_root_icbc = os.path.join(input_root, 'icbc')
 chem_prefix = 'cams_gqpe_'
 chem_nameformat = chem_prefix + '%Y%m%d_%H'
 chem_suffix = '.grb'
-chem_inc = 3
 
 icontools_runjobs = [
```
[the remark this answers is not in the document]
Comment (mjaehn) [reply cases/icon-art-oem-test/config.py:52]: Do you mean to replace icontools?

File: cases/icon-art-oem-test/config.py [hunk 7]
```diff
@@ -63,63 +57,56 @@
...
+    'cldopt_filename': ['rrtm_cldopt.nc', 'rad'],
+    'lrtm_filename': ['rrtmg_lw.nc', 'rad'],
+    'map_file_ana': ['map_file.ana', 'mapping'],
+    'chemtracer_xml_filename': ['tracers_oh_pntsrc.xml', 'XML'],
+    'pntSrc_xml_filename': ['pntSrc_example.xml', 'XML'],
+    'oem_gridded_emissions_nc': ['tno_3cat.nc', 'OEM'],
+    'oem_vertical_profiles_nc': ['vertical_profiles.nc', 'OEM'],
+    'oem_hourofday_nc': ['hourofday.nc', 'OEM'],
+    'oem_dayofweek_nc': ['dayofweek.nc', 'OEM'],
+    'oem_monthofyear_nc': ['monthofyear.nc', 'OEM'],
```
[the remark this answers is not in the document]
Comment (mjaehn) [reply cases/icon-art-oem-test/config.py:79]: True, but it could be adapted for COSMO to have it more uniform.

File: jobs/emissions.py (hunk 2 above)
Comment (mjaehn) [reply jobs/emissions.py:87]: I will check this. Could also be deleted by accident.

File: jobs/tools/write_cosmo_input_ghg.py (hunk 3 above)
Comment (mjaehn) [reply jobs/tools/write_cosmo_input_ghg.py:113]: Yes, this function is also called for ICON. I will rename it accordingly.

File: cases/icon-art-oem-test/config.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (mjaehn) [reply cases/icon-art-oem-test/config.py:52]: So these are case-specific bash scripts that could be replaced by python scripts. One could also be independent of `icontools` and use `cdo`. It is, however, quite a bit of work and probably beyond the scope of this task.

File: config/models.yaml (hunk 1 above)
Comment (mjaehn) [reply config/models.yaml:5]: `variants` and `features` change the behavior of the chain and are dependent on the used model. I discussed already with @gredvis and I think these two can be merged (i.e., only have `features`).

A job is a python script that is executed when running the chain. The jobs listed here are the default ones, i.e. if no jobs are specified, those are used.

File: jobs/tools/write_cosmo_input_ghg.py (hunk 3 above)
Comment (mjaehn) [reply jobs/tools/write_cosmo_input_ghg.py]: Ok apparently it is not used in ICON (anymore). I removed the parts within the function. The name can stay like that.

File: jobs/emissions.py (hunk 2 above)
Comment (mjaehn) [reply jobs/emissions.py]: I think it just worked well without these lines as the tests were successful locally and with jenkins. Anyways, I added this line again.

**Comment by mjaehn** [comment]
@efmkoene Thanks for your review. Indeed, the ensembles cases is not included here, but it will come back as a test case in the next iteration (see description of this PR). Let me know if you are happy with the changes.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (efmkoene) [reply README.md:23]: Oh OK, I'm in three projects so maybe there is no default one selected for me. Then it's OK!

File: cases/icon-art-oem-test/config.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (efmkoene) [reply cases/icon-art-oem-test/config.py:79]: Yeah, I have to say, it's actually quite neat. I'd support for COSMO following a similar quickfire settings procedure. If @gredvis agrees.

File: cases/icon-art-oem-test/config.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (efmkoene) [reply cases/icon-art-oem-test/config.py:52]: Absolutely. That'll be a different C2SM task (but a very useful one, I think)!

**Comment by efmkoene** [comment]
I'm happy with the changes. I'll let @gredvis and @jthanwer answer the outstanding questions/clarifications; either or both can then approve the PR.

File: cases/icon-art-oem-test/config.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (gredvis) [reply cases/icon-art-oem-test/config.py:79]: Yes, I agree that it would be nice to have the same look-and-feel for COSMO

**Comment by gredvis** [comment]
I briefly had a look at it together with Erik. Looks all good to me. I propose to move forward with the merge.

**Comment by gredvis** [comment]
Sorry, I should not have closed the pull request

File: cases/icon-art-global-test/config.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (gredvis) [reply cases/icon-art-global-test/config.py:66]: Does species2nudge have any impact on the results or is this is only a feature to be implemented in the future?

**Review by gredvis — APPROVED** [review]
Looks all good. Thanks for changing OAE to OEM at all places.

**Pull request merged**